common/src/api/internal/nexus.rs

```diff
@@ -103,3 +103,19 @@ pub struct UpdateArtifact {
 pub enum UpdateArtifactKind {
     Zone,
 }
+
+/// Sent to a sled agent to request the contents of an Instance's serial console.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema, PartialEq)]
+pub struct InstanceSerialConsoleRequest {
+    /// Character index in the serial buffer (since instance boot) from which to read.
+    pub byte_offset: Option<isize>,
+    /// Maximum number of bytes of buffered serial console contents (after byte_offset) to return.
+    pub max_bytes: Option<usize>,
+}
+
+/// Contents of an Instance's serial console buffer.
+#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
+pub struct InstanceSerialConsoleData {
+    pub data: Vec<u8>, // might not be UTF-8.
+    pub last_byte_offset: usize,
+}
```

Comment:
`InstanceSerialConsoleRequest` ends up being used by both Sled Agent's internal API and Nexus's external API, so `common/api/internal/nexus.rs` is an uncomfortable spot for it. Since it's so simple and doesn't depend on any other structs (and there's no essential requirement that it be identical between Nexus and Sled Agent) I lean toward giving Nexus and Sled Agent each their own copy in these two files:

https://github.com/oxidecomputer/omicron/blob/main/nexus/src/external_api/params.rs
https://github.com/oxidecomputer/omicron/blob/main/sled-agent/src/params.rs

`InstanceSerialConsoleData` is used in more spots and it's a bit harder for me to picture how it should look, but it probably could use the same treatment. Then when, for example, Nexus needs to refer to Sled Agent's response type, it can refer to the version on the generated Sled Agent client rather than the actual original struct used by Sled Agent.